website/src/components/docs/library/api/schemaTypes.ts

```diff
@@ -0,0 +1,136 @@
+const kindStrings = {
+  Module: null,
+  Interface: null,
+  Property: null,
+  Function: null,
+  "Call signature": null,
+  "Type alias": null
+};
+type KindString = keyof typeof kindStrings;
+
+export const isKindString = (input: string): input is KindString => {
+  return typeof input === "string" && kindStrings[input as KindString] !== undefined;
+};
+
+type BaseNode = {
+  name: string;
+  flags: {
+    isOptional?: boolean;
+  };
+};
+
+export type ModuleNode = BaseNode &
+  ParentNode & {
+    kindString: "Module";
+    children: (InterfaceNode | FunctionNode | TypeAliasNode)[];
+    sources: Source[];
+  };
+
+export type FunctionNode = BaseNode & {
+  kindString: "Function";
+  sources: Source[];
+  signatures: SignatureNode[];
+};
+
+export type SignatureNode = BaseNode & {
+  kindString: "Call signature";
+  comment?: Comment;
+  parameters: ParameterNode[];
+  type: AnyType;
+};
+
+export type ParameterNode = BaseNode & {
+  kindString: "Parameter";
+  comment?: Comment;
+  type: AnyType;
+};
+
+export type TypeAliasNode = BaseNode & {
+  kindString: "Type alias";
+  comment?: Comment;
+  sources: Source[];
+  type: AnyType;
+};
+
+export type InterfaceNode = BaseNode & {
+  kindString: "Interface";
+  children?: PropertyNode[];
+};
+
+export type PropertyNode = BaseNode & {
+  kindString: "Property";
+  comment?: Comment;
+  sources: Source[];
+  type: AnyType;
+};
+
+////---------
```

Comment:
Think it's been a while since I looked at this file (probs when we originally wrote it) - do we know what this comment line denotes? Might be worth sticking a comment / label on it if it meaningfully segments the file somehow, Otherwise I'm happy to remove it if it's not adding anything?